src/test/ui/consts/control-flow/short-circuit-let.rs

```diff
@@ -0,0 +1,18 @@
+// `&&` and `||` were previously forbidden in constants alongside let bindings.
+
+// run-pass
+
+#![feature(const_if_match)]
+
+const _: bool = {
+    let x = true && false || true;
+    x
+};
+
+const _: i32 = {
+    let mut x = 0;
+    let _  = true && { x = 1; false };
+    x
+};
```

Comment:
Let's also test the values at compile time & run-time.

Would be also good to have e.g.
```rust
const fn foo() -> bool {
    let x = true && return false; // and similar for `||`
    x
}
```